pkg/consuldp/lifecycle.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package consuldp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul-dataplane/pkg/envoy"
+)
+
+const (
+	// defaultLifecycleBindPort is the port which will serve the proxy lifecycle HTTP
+	// endpoints on the loopback interface.
+	defaultLifecycleBindPort = "20300"
+	cdpLifecycleBindAddr     = "127.0.0.1"
+	cdpLifecycleUrl          = "http://" + cdpLifecycleBindAddr
+
+	defaultLifecycleShutdownPath = "/shutdown"
+)
```

Comment:
I _think_ these constants shouldn't actually be needed and I have a bit of a config plumbing thing that needs fixing, because these defaults _should_ be coming from the the CLI configuration defaults in `cmd/consul-dataplane/main.go`